FOCUS #182: Usage Quantity (#186)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/dimensions/usagequantity.md

# Usage Quantity

> Usage Quantity is the amount of resources or services consumed. The Usage Quantity is measured in units specified by the Usage Unit. Usage Quantity may differ from a Pricing Quantity when providers use a different unit of measure for calculating cost.Usage Quantity is often used when analyzing environments to gain understanding of usage/consumption of given products/services in an environment.

Reviewer: ```suggestion
Usage Quantity is the amount of resources or services consumed. The Usage Quantity is measured in units specified by the Usage Unit. Usage Quantity may differ from a Pricing Quantity when providers use a different unit of measure for calculating cost. Usage Quantity is often used when analyzing environments to gain understanding of usage or consumption of a given product/service in an environment.

```

> Usage Quantity is the amount of resources or services consumed. The Usage Quantity is measured in units specified by the Usage Unit. Usage Quantity may differ from a Pricing Quantity when providers use a different unit of measure for calculating cost.Usage Quantity is often used when analyzing environments to gain understanding of usage/consumption of given products/services in an environment.

Reviewer 2: Usage Quantity is the amount of resources or services consumed. The Usage Quantity is measured in units specified by the Usage Unit. Usage Quantity may differ from a Pricing Quantity when providers use a different unit of measure for calculating cost. Usage Quantity is often used when analyzing environments to gain understanding of usage/consumption of given products/services in an environment.

> # Usage Quantity

Reviewer: The name should be updated to QuantityInUsageUnit to be consistent with other names.

The Usage Quantity represents the volume of a given resource and/or service used or purchased based on the [Usage Unit](#usageunit). Usage Quantity is often derived at a finer granularity or over a different time interval when compared to the [Pricing Quantity](#pricingquantity) (complementary to [Pricing Measurement Unit](#pricingmeasurementunit)), and focuses on resource and service consumption, not pricing and cost.

> UsageQuantity MUST be present in the billing data. This column MUST be a numeric value of type Decimal. The value MAY be negative in cases where [Charge Subcategory](#chargesubcategory) is 'Refund'. This column MUST NOT contain null values when [SKU Price ID](#skupriceid) is not null.

Reviewer 3: ```suggestion
UsageQuantity MUST be present in the billing data. This column MUST be a numeric value of type Decimal. The value MAY be negative in cases where [ChargeSubcategory](#chargesubcategory) is 'Refund'. This column MUST NOT contain null values when [SkuPriceId](#skupriceid) is not null.
```

# Usage Quantity

Usage Quantity represents the volume of a resource or service that was consumed, measured in units specified by the [UsageUnit](#usageunit). Distinct from [Pricing Quantity](#pricingquantity), Usage Quantity is used in conjunction with Usage Unit to measure the consumption of a product/service offering from a provider.

> The Usage Quantity for a charge may differ from Pricing Quantity when providers use different quantity, quantity increments, or columns to determine cost. The values in UsageUnit and UsageQuantity are often listed at a finer granularity or over a different time interval than the [PricingMeasure](#pricingmeasure) and [PricingQuantity](#pricingquantity). While UsageQuantity may appear to relate to pricing and cost, UsageQuantity is focused on resource consumption. It is essential not to confuse the UsageQuantity column with the PricingQuantity column which is the basis for determining cost. The UsageQuantity column MUST NOT be used as the basis for determining values related to any pricing or cost metric.

Reviewer 4: ```suggestion
The Usage Quantity for a charge may differ from Pricing Quantity when providers use different quantity, quantity increments, or columns to determine cost. The values in UsageUnit and UsageQuantity are often listed at a finer granularity or over a different time interval than the [PricingMeasurementUnit](#pricingmeasurementunit) and [PricingQuantity](#pricingquantity). While UsageQuantity may appear to relate to pricing and cost, UsageQuantity is focused on resource consumption. It is essential not to confuse the UsageQuantity column with the PricingQuantity column which is the basis for determining cost. The UsageQuantity column MUST NOT be used as the basis for determining values related to any pricing or cost metric.
```

UsageQuantity MUST exist in billing data. The column MUST be a numeric value. When [ChargeCategory](#chargecategory) is "Usage" or "Purchase", UsageQuantity MUST NOT be null or empty. When ChargeType is "Adjustment" and the adjustment applies to specific charges that had a common UsageUnit, UsageQuantity MUST NOT be null. 

## Column ID

UsageQuantity

## Display name

Usage Quantity

## Description

> The number of units of a resource or service that was used or consumed based on the [UsageUnit](#usageunit)

Reviewer 4: ```suggestion
Number of units of a resource or service that was used or consumed based on the [UsageUnit](#usageunit).
```
